src/Backend/DataSeriesFactory.js

```diff
@@ -0,0 +1,27 @@
+export default class TraceFactory{
+    static getXAxisByPoints(start, end, points){
+        const interval = (end-start)/points
+        return this.getXAxis(start,end, points,interval)
+    }
+
+    static getXAxisByInterval(start,end,interval){
+        const points = Math.ceil((end-start)/interval)
+        return this.getXAxis(start,end,points,interval)
+    }
+
+    static getXAxis(start,end,points,interval){
+        console.log(points)
+        const arr = Array.from(new Array(points-1),(_,key)=>{return (Number)((start + interval * (key+1).toFixed(3)))})
```

Comment:
Explain the logic here. I am especially worried about `toFixed(3)` - why is this added ? Won't this lead to a different numerical results than calculations with full precision ?

Reply:
It was added to prevent the floating point error from being a factor in the stp calc. Sadly it didn't work and i forgot to delete it